trikot-analytics/firebase-ktx/src/main/java/com/mirego/trikot/analytics/FirebaseAnalyticsService.kt

```diff
@@ -40,6 +49,23 @@ class FirebaseAnalyticsService(context: Context, analyticsEnabled: Boolean = tru
             }
     }
 
+    override suspend fun distinctDeviceId() = suspendCancellableCoroutine { continuation ->
+        firebaseAnalytics.appInstanceId
+            .addOnCompleteListener { task ->
+                val result = task.result
+                if (task.isSuccessful && result != null) {
+                    continuation.resume(result)
+                } else if (!task.isSuccessful) {
+                    continuation.cancel(Throwable(task.exception))
```

Comment:
Typed exceptions might help catching specific errors here.

also, why not resumeWithException